server-phoenix/test/helios_web/controllers/web_hooks/slack_controller_test.exs

```diff
@@ -0,0 +1,163 @@
+          event: %{
+            client_msg_id: "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+            type: "message",
+            text: "hi",
+            user: "Priscila",
+            ts: "155984192.001600",
+            channel: "CXXXXXXXX",
+            event_ts: "1559841925.001600",
+            channel_type: "channel"
+          },
+          type: "event_callback",
+          event_id: "<%= @event_id %>",
+          event_time: 1_559_841_923,
+          authed_users: ["UXXXXXXXX", "UXXXXXXXX"],
+          slack: %{
+            token: "Sim-123456",
+            team_id: "TXXXXXXXX",
+            api_app_id: "AXXXXXX",
+            event: %{
+              client_msg_id: "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+              type: "message",
+              text: "hi",
+              user: "UXXXXXXXX",
+              ts: "1559841924.001600",
+              channel: "CXXXXXXXX",
+              event_ts: "1559841924.001600",
+              channel_type: "channel"
+            },
+            type: "event_callback",
+            event_id: "EvXXXXXX",
+            event_time: 1_559_841_923,
+            authed_users: ["UXXXXXXXX", "UXXXXXXXX"]
+          }
+        }
+      )
+
+    messages = get_slack_messages()
+    assert length(messages) == 1
+  end
+
+  test "assert an author was populated in the db", %{conn: conn} do
+    conn =
+      conn
+      |> put_req_header("content-type", "application/json")
+      |> post(
+        Routes.slack_path(conn, :handle),
+        %{
+          token: "Sim-123456",
+          team_id: "TXXXXXXXX",
+          api_app_id: "AXXXXXXXX",
+          event: %{
+            client_msg_id: "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+            type: "message",
+            text: "hi",
+            user: "Priscila",
+            ts: "155984192.001600",
+            channel: "CXXXXXXXX",
+            event_ts: "1559841925.001600",
+            channel_type: "channel"
+          },
+          type: "event_callback",
+          event_id: "<%= @event_id %>",
+          event_time: 1_559_841_923,
+          authed_users: ["UXXXXXXXX", "UXXXXXXXX"],
+          slack: %{
+            token: "Sim-123456",
+            team_id: "TXXXXXXXX",
+            api_app_id: "AXXXXXX",
+            event: %{
+              client_msg_id: "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+              type: "message",
+              text: "hi",
+              user: "UXXXXXXXX",
+              ts: "1559841924.001600",
+              channel: "CXXXXXXXX",
+              event_ts: "1559841924.001600",
+              channel_type: "channel"
+            },
+            type: "event_callback",
+            event_id: "EvXXXXXX",
+            event_time: 1_559_841_923,
+            authed_users: ["UXXXXXXXX", "UXXXXXXXX"]
+          }
+        }
+      )
+
+    messages = get_slack_messages()
+    first_message = messages |> Enum.at(0) |> IO.inspect()
+    assert Map.fetch(first_message, :source_author) == {:ok, "Priscila"}
```

Comment:
Its looks like this assertion is the only difference from `"check for data in table",`. I'm ok with multiple assertions in the test e.g. a row must be present before you assert the author is set. If you want to keep the separate tests, you could extract a setup method:
```
describe "when a slack web hook event is submitted" do
  setup [:post_slack_message]

  test ...

  test ...
end

defp post_slack_message(%{conn: conn}) do
  ...
end
```

Reply:
Good point. Combined those two tests into one.